plugin/cache.go

```diff
@@ -98,14 +98,43 @@ func ClearCacheFor(path string, clearParentList bool) []string {
 	deleted := cache.Delete(rx)
 
 	if clearParentList {
+		// If an immediate parent exists, clear that. Otherwise search for the first available ancestor
```

Comment:
If we change `clearParentList` => `clearAncestorList`, then could this be simplified to
```
ancestor = parent
if entry.Prefetched() {
  ancestor = getAncestor...
}
// Clearing logic here
```

Reply:
We don't have an entry, and I want to avoid getting one because that requires doing a List call that might not be cached.

Reply:
Ah I gotcha, I missed the fact that we didn't have an entry. But couldn't we still capture the basic idea in the current code? It would just be:

```
sourceAncestorID := getSourceAncestorPathFromCache(path)
// Clear the sourceAncestor's list op from the cache
...
```

(I'm using sourceAncestor because that's how we word things in other places, like the[ fuse code](https://github.com/puppetlabs/wash/blob/master/fuse/core.go#L119))

Also, I do think `clearParentList` => `clearSourceAncestorList` (since the issue shows that we may not clear parent's list for _all_ entries).

Reply:
I'll try. I was having trouble making it generic.